wasm-node/javascript/src/index-deno.ts

```diff
@@ -224,7 +228,7 @@ function connect(config: ConnectionConfig, forbidTcp: boolean, forbidWs: boolean
             } else {
                 // TCP
                 connection.socket.destroyed = true;
-                connection.socket.inner.then((connec) => connec.close());
+                connection.socket.inner.then((connec) => connec?.close());
```

Comment:
Can `connec` actually be null/undefined here? I feel like the only way that this could happen is if `reset` was called twice, which would be a smoldot bug

Reply:
I returned `null` when `Deno.connect` returns error, so added this null-check

Reply:
Ah I see, I had to re-read the documentation of `Promise.catch` to understand

I'm trying very hard to not lose track of what's happening, which can very easily happen with JavaScript.
Given that `reset` should never be called after `onConnectionReset`, this should never be `null`, so:

```suggestion
                connection.socket.inner.then((connec) => connec!.close());
```